scripts/name_match.py

```diff
@@ -0,0 +1,177 @@
+        elif cleaned_name.replace("'", "") == cleaned_person_family_name.replace("'", ""):
+            # Example O'Donnell
+            matched = True
+        elif cleaned_name.replace(" ", "") == cleaned_person_family_name.replace(" ", ""):
+            # Example Crosswhite Hader == CrosswhiteHader
+            matched = True
+    elif cleaned_name == cleaned_person_name:
+        matched = True
+    elif cleaned_name in cleaned_person_name:
+        matched = True
+    elif (" of " in cleaned_name) and (cleaned_name.split()[0] == cleaned_person_name.split()[-1]):
+        # Example: carpenter of aroostook
+        matched = True
+    elif cleaned_name.split()[0] == cleaned_person_name.split()[-1]: # Need to double check tests
+        # Another example for: Carpenter of Aroostook,
+        matched = True
+    elif len(cleaned_name.split()) == 2 and (cleaned_name.split()[0] == cleaned_person_name.split()[0]) and (cleaned_name.split()[-1] != cleaned_person_name.split()[-1]):
+        # 'Mark BLIER' and 'Mark Lawrence'
+        matched = False
+    elif len(cleaned_name.split()) > 0 and (cleaned_name.split()[0] == cleaned_person_name.split()[-1]):
+        # Example: West (Tammy)
+        matched = True
+    elif len(cleaned_name.split()) == 2 and (cleaned_name.split()[1] in cleaned_person_name):
+        #Matt Huffman, M.
+        matched = True
+    elif len(cleaned_name.split()) > 2 and (cleaned_name.split()[0] in cleaned_person_name):
+        # 'SANBORN, L. of Cumberland' and 'Victoria E. Morales'
+        matched = True
+
+    if matched:
+        return True
+    else:
+        return False
+
+
+@click.command()
+@click.argument("archive_data_csv")
+def entrypoint(archive_data_csv):
+    archive_data = []
+    jurisdiction = ""
+    with open(archive_data_csv, newline='') as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            archive_data.append({
+                "name": row['name'],
+                "jurisdiction": row['jurisdiction'],
+                "session": row['session'],
+                "num_occurances": row['num_occurances']
+            })
+
+    jurisdiction = archive_data[0]["jurisdiction"]
+    existing_people = []
+    for filename in glob.glob(os.path.join(get_data_dir(archive_data[0]["jurisdiction"]), "people/*.yml")):
+        with open(filename) as f:
+            existing_people.append(load_yaml(f))
+
+    for line in archive_data:
+        for person in existing_people:
+            matched = find_match(line["name"], person)
+            if matched:
+                match_confirmed = interactive_check(line["name"], person["name"], matched)
+                if match_confirmed:
+                    person_file = get_filename(person)
+                    output_dir = get_data_dir(jurisdiction)
+                    person_file = os.path.join(os.path.join(output_dir, "people"), person_file)
+                    add_to_file(line["name"], person_file)
+                break
```

Comment:
is this break in the correct spot?  I'd assume it would only break if match is confirmed